coderd/userauth.go

```diff
@@ -81,6 +91,9 @@ func (api *API) userOAuth2Github(rw http.ResponseWriter, r *http.Request) {
 				Message: "Internal error fetching authenticated Github user organizations.",
 				Detail:  err.Error(),
 			})
+			// We pass a disposable user ID just to force an audit diff
+			// and generate a log for a failed login
+			aReq.New = database.APIKey{UserID: uuid.New()}
```

Comment:
Can we make a reusable thing for this?

Either something like: 

```golang
var AnonymousUser = uuid.MustParse("11111111-1111-1111-1111-111111111111")
```

or maybe we generate some hash from the user IP so their ID is deterministic if it's the same person each time. Or we store some id on the client for the anon user.

I don't know for sure. This is probably ok for now, but we can make this better.